contrib/!lang/typescript/packages.el

```diff
@@ -4,20 +4,62 @@
 ;; Copyright (c) 2014-2015 Chris Bowdon & Contributors
 ;;
 ;; Author: Chris Bowdon <c.bowdon@bath.edu>
+;; Author: Hamish Hutchings <hamhut1066@gmail.com>
 ;; URL: https://github.com/syl20bnr/spacemacs
 ;;
 ;; This file is not part of GNU Emacs.
 ;;
 ;;; License: GPLv3
 
-(setq typescript-packages '(tss))
+(setq typescript-private-packages '(
+                                    tss
+                                    flycheck))
 
-(defun typescript/init-tss ()
-  "Initialize my package"
+(setq test-layer-excluded-packages '(flymake))
+
+(defun typescript-private/post-init-flycheck ()
+  "Add hook for flycheck"
+  (use-package flycheck
+    :defer t
+    :config (progn
+              ;; ;;; remove hook for flymake, as it seems to conflict.
+              ;; (add-hook 'typescript-mode-hook 'flycheck-mode (flymake-mode nil))
+              ;; (add-hook 'typescript-mode-hook (lambda flymake-mode nil))
+              ;;; This Code Snippet was from [Here](https://github.com/caisah/flycheck-typescript) and is licensed under the GPL
+              (flycheck-define-checker typescript
+                "A TypeScript syntax checker using tsc command."
+                :command ("tsc" "--out" "/dev/null" source-inplace)
+                :error-patterns
+                ((error line-start (file-name) "(" line "," column "): error " (message) line-end))
+                :modes (typescript-mode))
+
+              (add-to-list 'flycheck-checkers 'typescript)
+              )
+    )
+  )
+
```

Comment:
Its better to keep trailing parentheses together, don't leave them on their own line